app/lib/tool/maintenance/update_public_bucket.dart

```diff
@@ -38,6 +54,53 @@ Future<int> updatePublicArchiveBucket() async {
       );
       updated++;
     }
+    objectNamesInPublicBucket.add(objectName);
+  }
+
+  await for (final entry in publicBucket.list()) {
+    // Skip non-objects.
+    if (!entry.isObject) {
+      continue;
+    }
+    // Skip objects that were matched in the previous step.
+    if (objectNamesInPublicBucket.contains(entry.name)) {
+      continue;
+    }
+
+    final publicInfo = await publicBucket.tryInfo(entry.name);
+    if (publicInfo == null) {
+      _logger.warning(
+          'Failed to get info for public bucket object "${entry.name}".');
+      continue;
+    }
+
+    // Skip recently updated objects.
+    if (publicInfo.age < ageCheckThreshold) {
+      // Ignore recent files.
+      continue;
+    }
+
+    final canonicalInfo = await canonicalBucket.tryInfo(entry.name);
+    if (canonicalInfo != null) {
+      // Warn if both the canonical and the public bucket has the same object,
+      // but it wasn't matched through the [PackageVersion] query above.
+      if (publicInfo.age < ageCheckThreshold) {
+        // Ignore recent files.
+        continue;
+      }
```

Comment:
Sure you don't want this?
```suggestion
      if (canonicalInfo.age < ageCheckThreshold) {
        // Ignore recent files.
        continue;
      }
```

Reply:
Ah, good catch! thanks!